techmatters/hrm
# HRM support for async jobs II: Handle possible failures in completed jobs

Primary reviewer: @stephenhand 

## Description
This PR is a follow up of https://github.com/techmatters/hrm/pull/239.
Here I'm including:
- Handling jobs that are "completed" but with a failure. In this case we also keep track of the payloads on failure.
- Adding max retries so the jobs won't be infinitely attempted. TODO: Exponential backoff.

### Checklist

### Verification steps
- Trust the tests? :P or

- Start a local server, expose via ngrok, point to it from Flex and save a chat contact.
- Confirm that a new pending job was created, and that it is published on the next iteration of the processor.
- Stop the server and "mock" the polling function to return the same contact job as before, with the same shape as it was published, plus an `attemptNumber === 1`, `attemptResult === 'failed'` and `attemptPayload === <error description>`.
- Run the server again and check that the "faked completed job" is processed accordingly, adding the above payload to the key `1` of the `failedAttemptsPayloads` column.
- Repeat the above two steps with `attemptNumber === 20` and confirm that on top of the above, the job is marked as complete with a failure message.
